### PR TITLE
[RFR] Storm 977/rename actionexec to liveaction

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
## TODO: 

- [X] - Merge https://github.com/StackStorm/st2/pull/1014 when it's done

- [x] - Remove separate history process - not needed anymore
 - [x] - Write utility code to update execution model and liveaction model in one go.  [https://github.com/StackStorm/st2/pull/1018]
 - [x] - Move st2common/util/executions.py st2common/service/executions.py
 - [x] - POST /executions should create both live action and actionexecution db models
 - [x] - Update execution model when results come back from runners (use utility above)
 - [x] - Update execution model when querier gets results (use utility above)

- [x] Rename 'execution' field in ActionExecution (previously History) to 'liveaction'

- [x] - One box testing. Focus on st2 execution * commands
- [x] - st2client/st2cli might be broken
- [x] - st2web might be broken 
- [X] - Migration tool for people using versions <= 0.8? Decided: No migration support. 
- [x] - Docs 
- [x] - Changelog

### CODE REVIEW

**Comment by manasdk** [comment]
@lakshmi-kannan Should we hide liveactions from the API? e.g. 

POST @ /execution should take an ActionExecution instead of LiveAction - right?

**Comment by manasdk** [comment]
Looking at the entire PR it seems like it is better to continue with 2 different endpoints for now.

**Comment by lakshmi-kannan** [comment]
@manasdk Yes, LiveActions should be completely hidden from all APIs.

**Comment by lakshmi-kannan** [comment]
>POST @ /execution should take an ActionExecution instead of LiveAction - right?

It takes in a weird object today. It cannot take in an ActionExecution object because live action is really what user needs to specify. I think it's fine to leave this as is. The response to POST can be ActionExecution.

**Comment by manasdk** [comment]
>  The response to POST can be ActionExecution.

Argh. Is it ok for POST response to be a different object from the one POSTed? Does that lead to surprise. If we all agree I am happy to change.

**Comment by Kami** [comment]
@manasdk how is this change going?

Are you OK with @enykeev or me putting mistral task name attribute in the `context` field of the `LiveAction` model so we can display mistral stuff in the UI?

**Comment by manasdk** [comment]
Update st2web and st2client.js

https://github.com/StackStorm/st2web/pull/88
https://github.com/StackStorm/st2client.js/pull/11

**Comment by manasdk** [comment]
Working on docs and changelog.

**Comment by manasdk** [comment]
Net gain with the PR ->
+1,288  −1,124 = 164

Considering how much time has been spent on this thing already feels rather meager. :(

**Comment by manasdk** [comment]
AFAICT docs are already fixed.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -33,6 +33,12 @@ Docs: http://docks.stackstorm.com/latest
   being executed. (new-feature)
 * Cast params of an execution before scheduling in the RulesEngine. This allows non-string
   parameters in an action. (new-feature)
+* Rename ActionExecution to LiveAction. (refactor)
+* Rename ActionExecutionHistor to ActionExecution. (refactor)
```
Comment (m4dcoder) [inline CHANGELOG.rst]: misspelled ActionExecutionHistory

File: README.md [hunk 2]
```diff
@@ -39,7 +39,7 @@ StackStorm plugs into the environment via the extensible set of adapters: sensor
 
 * **Packs** are the units of content deployment. They simplify the management and sharing of StackStorm pluggable content by grouping integrations (triggers and actions) and automations (rules and workflows). A growing number of packs is available on StackStorm community. User can create their own packs,  share them on Github, or submit to StackStorm community repo.
 
-* **Audit trail** of action executions, manual or automated, is recorded and stored with full details of triggering context and execution results. It is is also captured in audit logs for integrating with external logging and analytical tools: LogStash, Splunk, statsd, syslog.
+* **Audit trail** of action liveactions, manual or automated, is recorded and stored with full details of triggering context and execution results. It is is also captured in audit logs for integrating with external logging and analytical tools: LogStash, Splunk, statsd, syslog.
```
Comment (m4dcoder) [inline README.md]: "Audit trail of action executions" sounds better.

File: docs/source/history.rst [hunk 3]
```diff
@@ -1,7 +1,7 @@
 History and Audit
 =================
 
-These are records maintained by |st2| of past action executions.  These records include information about what events triggered what rules and then what actions; common uses for this information includes root cause analysis reporting as well as operational control as well as collaboration.
+These are records maintained by |st2| of past action liveactions.  These records include information about what events triggered what rules and then what actions; common uses for this information includes root cause analysis reporting as well as operational control as well as collaboration.
```
Comment (m4dcoder) [inline docs/source/history.rst]: "These are records maintained by |st2| of past action executions" seems more readable.

File: docs/source/history.rst [hunk 4]
```diff
@@ -17,7 +17,7 @@ All execution records are accessible by the ``st2 execution *`` command family.
 
 .. code-block:: bash
 
-    # to see last 5 executions
+    # to see last 5 liveactions
```
Comment (m4dcoder) [inline docs/source/history.rst]: put a space between live actions in UI/CLI output?  i.e. "to see last 5 live actions"

File: docs/source/overview.rst [hunk 5]
```diff
@@ -22,26 +22,26 @@ How it works
...
- 
+
 * **Packs** are the units of content deployment. They simplify the management and sharing of |st2| pluggable content by grouping integrations (triggers and actions) and automations (rules and workflows). A growing number of packs is available on |st2| community. User can create their own packs,  share them on Github, or submit to |st2| community repo.
 
-* **Audit trail** of action executions, manual or automated, is recorded and stored with full details of triggering context and execution results. It is is also captured in audit logs for integrating with external logging and analytical tools: LogStash, Splunk, statsd, syslog.
+* **Audit trail** of action liveactions, manual or automated, is recorded and stored with full details of triggering context and execution results. It is is also captured in audit logs for integrating with external logging and analytical tools: LogStash, Splunk, statsd, syslog.
```
Comment (m4dcoder) [inline docs/source/overview.rst]: Same comments above on readability.

File: docs/source/roadmap.rst [hunk 6]
```diff
@@ -4,18 +4,18 @@ Roadmap
 StackStorm is new and under active development. We are opening it early to engage community, get  feedback, and refine directions, and welcome contributions. Below are key next items we see as top priorities.
 
 
-* **Web UI:** refactor history view, create and edit rules and workflows, add graphical representations for workflow definitions and executions.
-* **Improving** `Mistal <https://wiki.openstack.org/wiki/Mistral>`_  **integration:** focus on simplifying Mistral DSL for |st2| actions, visibility of workflow executions, and reliability of |st2|-Mistral communication.
+* **Web UI:** refactor history view, create and edit rules and workflows, add graphical representations for workflow definitions and liveactions.
```
Comment (m4dcoder) [inline docs/source/roadmap.rst]: executions read better

File: docs/source/roadmap.rst [hunk 7]
```diff
@@ -4,18 +4,18 @@ Roadmap
...
+* **Improving** `Mistal <https://wiki.openstack.org/wiki/Mistral>`_  **integration:** focus on simplifying Mistral DSL for |st2| actions, visibility of workflow liveactions, and reliability of |st2|-Mistral communication.
```
Comment (m4dcoder) [inline docs/source/roadmap.rst]: workflow executions read better

File: docs/source/start.rst [hunk 8]
```diff
@@ -94,11 +94,11 @@ as described in :ref:`config-configure-ssh` section.
     # Crasily complex command passed with `cmd`
     st2 run core.remote hosts='localhost' cmd="for u in bob phill luke; do echo \"Logins by \$u per day:\"; grep \$u /var/log/secure | grep opened | awk '{print \$1 \"-\" \$2}' | uniq -c | sort; done;"
 
-Check the action execution history and details of action executions with ``st2 execution`` command:
+Check the action execution history and details of action liveactions with ``st2 execution`` command:
```
Comment (m4dcoder) [inline docs/source/start.rst]: "details of live actions"?

File: docs/source/workflows.rst [hunk 9]
```diff
@@ -4,7 +4,7 @@ Workflows
 Typical datacenter operations and processes involve taking multiple actions
 across various systems. To capture and automate these operations,
 |st2| uses workflows. Workflow strings atomic actions into a higher level
-automation, and orchestrates their executions by calling the right action
+automation, and orchestrates their liveactions by calling the right action
```
Comment (m4dcoder) [inline docs/source/workflows.rst]: "executions" read better

File: st2actions/st2actions/runners/mistral/v2.py [hunk 10]
```diff
@@ -147,7 +147,7 @@ def run(self, action_parameters):
 
         if not is_workbook:
             # Non-workbook definition containing multiple workflows is not supported.
-            if len([k for k, v in six.iteritems(def_dict) if k != 'version']) != 1:
+            if len([k for k, _ in six.iteritems(def_dict) if k != 'version']) != 1:
```
Comment (m4dcoder) [inline st2actions/st2actions/runners/mistral/v2.py]: How is this line change related to this PR?

Comment (manasdk) [inline st2actions/st2actions/runners/mistral/v2.py]: local linter complained so I fixed it. It would seem that my linter is more strict in some regards than `make flake8`. 

To be clear this change is unrelated.

**Comment by manasdk** [comment]
@m4dcoder Docs updates in e53c41b3dff9a8088730c0447d1e3f5b5ed71b19
